### ISSUE TITLE
Swift Package unable to find headers

### ISSUE THREAD

**Issue by pixlwave** [0xced/XCDYouTubeKit#441]
I'm not sure if this is a toolchain issue or a project issue, but it seems #430 no longer works correctly. I've been trying it within Xcode 11 but to make it easier to debug, the following is the output of a command line build:

``` bash
mac:~ dev$ git clone https://github.com/0xced/XCDYouTubeKit
Cloning into 'XCDYouTubeKit'...
remote: Enumerating objects: 141, done.
remote: Counting objects: 100% (141/141), done.
remote: Compressing objects: 100% (66/66), done.
remote: Total 7920 (delta 81), reused 112 (delta 72), pack-reused 7779
Receiving objects: 100% (7920/7920), 22.48 MiB | 6.36 MiB/s, done.
Resolving deltas: 100% (5137/5137), done.
mac:~ dev$ cd XCDYouTubeKit/
mac:XCDYouTubeKit dev$ swift build
warning: no include directory found for target 'XCDYouTubeKit'; libraries cannotIn file included from /Users/dev/XCDYouTubeKit/XCDYouTubeKit/XCDYouTubeVideo.m:5:
/Users/dev/XCDYouTubeKit/XCDYouTubeKit/XCDYouTubeVideo+Private.h:5:9: fatal error: 'XCDYouTubeKit/XCDYouTubeVideo.h' file not found
#import <XCDYouTubeKit/XCDYouTubeVideo.h>
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Users/dev/XCDYouTubeKit/XCDYouTubeKit/XCDYouTubeClient.m:5:
/Users/dev/XCDYouTubeKit/XCDYouTubeKit/XCDYouTubeClient.h:14:9: fatal error: 'XCDYouTubeKit/XCDYouTubeOperation.h' file not found
#import <XCDYouTubeKit/XCDYouTubeOperation.h>
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Users/dev/XCDYouTubeKit/XCDYouTubeKit/XCDYouTubeVideoOperation.m:5:
/Users/dev/XCDYouTubeKit/XCDYouTubeKit/XCDYouTubeVideoOperation.h:13:9: fatal error: 'XCDYouTubeKit/XCDYouTubeOperation.h' file not found
#import <XCDYouTubeKit/XCDYouTubeOperation.h>
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

mac:XCDYouTubeKit dev$ 
```

I'm building on macOS 10.14.5, with both Xcode 10 and 11 beta 4 installed. `swift --version` reports `Apple Swift version 5.0.1 (swiftlang-1001.0.82.4 clang-1001.0.46.5)` so I assume in the command line at least it's still using the stable toolchain.

@Dschee Is this still working for you?

**Comment by Dschee** [comment]
@pixlwave Hey, I'm not sure how you're incorporating this library into your project. Are you using Xcode 11's built-in SwiftPM support? Or are you using Accio? Could you please (in both cases) post the contents of your `Package.swift` file? That way, I know what exactly to check for.

**Comment by pixlwave** [comment]
Hey @Dschee Sorry, I didn’t even know there was any other way to use Swift Packages in Xcode outside of the beta, otherwise I would have been more specific 🙂

I don’t have a Swift.package file. Simply clicked File | Swift Packages | Add Package Dependency... and entered the GitHub address, leaving the version at its default of 2.7.7 up to Next Major. Then clicked build and it fails to compile. Errors are the same as above from the command line, failing to find XCDYouTubeVideo.h and XCDYouTubeOperation.h.

**Comment by pixlwave** [comment]
I guess this is the closest representation to Package.swift in Xcode?

![4B3F08D6-BC16-480B-B7D7-2C68505984AC](https://user-images.githubusercontent.com/6060466/61654635-c3c3f180-acb4-11e9-9a28-4868a38afdb6.jpeg)

**Comment by Dschee** [comment]
@pixlwave Thanks for the clarification, that helps.

The header linking problem can be fixed easily within the dependency. For example instead of this ...
```swift
#import <XCDYouTubeKit/XCDYouTubeOperation.h>
#import <XCDYouTubeKit/XCDYouTubeVideo.h>
#import <XCDYouTubeKit/XCDYouTubeError.h>
```

... by simply writing this ...
```swift
#import "XCDYouTubeOperation.h"
#import "XCDYouTubeVideo.h"
#import "XCDYouTubeError.h"
```

... the build problem regarding finding the headers disappears. But that's only the start, once that's fixed, many additional build errors appear. Mostly they seem to be issues with missing iOS libraries, like `UIKit` and `MediaPlayer`. I'm not sure if the SwiftPM support within Xcode can handle libraries that depend on specific platform libraries – couldn't figure it out quickly.

One thing that is definitely needed for it to even potentially work is to update the `Package.swift` file to uncomment [the platform line](https://github.com/0xced/XCDYouTubeKit/blob/master/Package.swift#L6) and to update the manifest version to `5.0` (from `4.2`).

So, for now I can't add true SwiftPM support to this library. I will have a second look into this though when I have some more free time and report if I make any progress. In any case, at first glance, it seems SwiftPM support within Xcode doesn't completely replace Accio for now ...

**Comment by pixlwave** [comment]
> ... by simply writing this ...
> 
> ```objc
> #import "XCDYouTubeOperation.h"
> #import "XCDYouTubeVideo.h"
> #import "XCDYouTubeError.h"
> ```

Damn I just tried this again (tried before I opened the Issue) and it works. I think before I was using the command line running `swift build` where I still see the errors you are on about.
However, this time I did it all in Xcode 11 by opening the project using Package.swift, editing those `#import` statements, choosing iOS as a target and then it builds no problems whatsoever.

Is this a change that would break the framework in any way?

**Comment by BreeeZe** [comment]
I'm running into the same problem.
I had a bunch of packages referenced via CocoaPods and I wanted to see if I could switch them to Swift Packages.
I added XCDYouTubeKit to my project via the integrated SwiftPackageManager in XCode 11 and it doesn't compile giving me the above error.

I cloned the repo to my local repo and changed the references to make it compile, but if I add this "fixed" version to my project it can't find any classes or interfaces from the XCDYouTubeKit, although it accepts the "import XCDYouTubeKit" statement.

My Project is in swift 4 by the way.

**Comment by BreeeZe** [comment]
> > ... by simply writing this ...
> > ```objc
> > #import "XCDYouTubeOperation.h"
> > #import "XCDYouTubeVideo.h"
> > #import "XCDYouTubeError.h"
> > ```
> 
> Damn I just tried this again (tried before I opened the Issue) and it works. I think before I was using the command line running `swift build` where I still see the errors you are on about.
> However, this time I did it all in Xcode 11 by opening the project using Package.swift, editing those `#import` statements, choosing iOS as a target and then it builds no problems whatsoever.
> 
> Is this a change that would break the framework in any way?
> 
> **Edit**: Ok, scratch that! For some reason Xcode isn't actually building anything... Doesn't fail though 😉

I forked the repo and managed to get it to work, you can clone my repo:
https://github.com/BreeeZe/XCDYouTubeKit.git and choose the "develop" branch in the package manager.
I had to exclude the VideoPlayerViewController since I use this lib on tvOS, and could figure out if and how to include files conditionally in a package 🤔

**Comment by AndreiVidrasco** [comment]
@BreeeZe maybe you could open a PR back here?

**Comment by BreeeZe** [comment]
I did just now, but I get some errors concerning the conditional compilation statements I'll have to fix..

**Comment by SoneeJohn** [comment]
Closed with v2.8.2
